pkg/kclient/deployments.go

```diff
@@ -0,0 +1,43 @@
+package kclient
+
+import (
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// constants for deployments
+const (
+	DeploymentKind       = "Deployment"
+	DeploymentAPIVersion = "apps/v1"
+)
+
+// CreateDeployment creates a deployment based on the given pod
+func (c *Client) CreateDeployment(pod *corev1.Pod) (*appsv1.Deployment, error) {
+
+	replicas := int32(1)
+	deployment := appsv1.Deployment{
```

Comment:
Would it make sense to create a function for generating deployment spec as well? Something similar to `GenerateContainerSpec()` and `GeneratePodSpec()`

Reply:
Good point! I've created a `GenerateDeploymentSpec` function. So there is now both `GenerateDeploymentSpec` to create the spec and then `CreateDeployment` to create the actual deployment resource.

I've also updated the pod function to generate a `PodTemplateSpec` which can now be directly fed into the `GenerateDeploymentSpec` function.